src/views/tickets/view-link.php

```diff
@@ -21,92 +21,16 @@
-}
-
-$event = get_post( $event_id );
-
-if ( empty( $event ) ) {
-	return;
-}
-
-$post_type = get_post_type_object( $event->post_type );
-$user_id   = get_current_user_id();
-
-if ( empty( $post_type ) || ! is_user_logged_in() ) {
-	return;
-}
-
-/**
- * Filters the arg to determine whether or not to hide the view link.
- *
- * @since 5.3.2
- *
- * @param bool True/false to hide view link.
- */
-$hide_view_link = apply_filters( 'tec_tickets_hide_view_link', tribe_doing_shortcode( 'tribe_tickets' ) );
-if ( $hide_view_link ) {
-	return;
-}
-
-$is_event_page = class_exists( 'Tribe__Events__Main' ) && Tribe__Events__Main::POSTTYPE === $event->post_type;
-
-$post_type_singular = $post_type ? $post_type->labels->singular_name : _x( 'Post', 'fallback post type singular name', 'event-tickets' );
-$counters           = [];
-$rsvp_count         = $view->count_rsvp_attendees( $event_id, $user_id );
-$ticket_count       = $view->count_ticket_attendees( $event_id, $user_id );
-
-if ( 1 === $rsvp_count ) {
-	// Translators: 1: the number one, 2: singular RSVP label.
-	$counters[] = sprintf( _x( '%1$d %2$s', 'RSVP count singular', 'event-tickets' ), $rsvp_count, tribe_get_rsvp_label_singular( basename( __FILE__ ) ) );
-} elseif ( 1 < $rsvp_count ) {
-	// Translators: 1: the plural number of RSVPs, 2: plural RSVP label.
-	$counters[] = sprintf( _x( '%1$d %2$s', 'RSVP count plural', 'event-tickets' ), $rsvp_count, tribe_get_rsvp_label_plural( basename( __FILE__ ) ) );
-}
-
-if ( 1 === $ticket_count ) {
-	// Translators: 1: the number one, 2: singular Ticket label.
-	$counters[] = sprintf( _x( '%1$d %2$s', 'Ticket count singular', 'event-tickets' ), $ticket_count, tribe_get_ticket_label_singular( basename( __FILE__ ) ) );
-} elseif ( 1 < $ticket_count ) {
-	// Translators: 1: the plural number of Tickets, 2: plural Ticket label.
-	$counters[] = sprintf( _x( '%1$d %2$s', 'Ticket count plural', 'event-tickets' ), $ticket_count, tribe_get_ticket_label_plural( basename( __FILE__ ) ) );
-}
-
-if ( empty( $counters ) ) {
-	return false;
-}
-
-$link = $view->get_tickets_page_url( $event_id, $is_event_page );
-
-// Translators: 1: number of RSVPs and/or Tickets with accompanying ticket type text, 2: post type label.
-$message = esc_html( sprintf( __( 'You have %1s for this %2s.', 'event-tickets' ), implode( _x( ' and ', 'separator if there are both RSVPs and Tickets', 'event-tickets' ), $counters ), $post_type_singular ) );
-?>
-
-<div class="tribe-link-view-attendee">
-	<?php echo $message ?>
-	<a href="<?php echo esc_url( $link ); ?>">
-		<?php
-		// Translators: %s: The name(s) of the type(s) of ticket(s) the specified user (optional) has for the specified event.
-		echo sprintf( esc_html__( 'View your %s', 'event-tickets' ), $view->get_description_rsvp_ticket( $event_id, $user_id ) );
-		?>
-	</a>
-</div>
+/** @var Tribe__Tickets__Editor__Template $template */
+$template = tribe( 'tickets.editor.template' );
+$template->template( 'blocks/attendees/view-link' );
```

Comment:
Loading this file, and trigger all the attached filtering, might be skipped simply including the `blocks/attendees/view-link` template directly in the file that would include this one. Why keep including this file?

Reply:
As discussed above, this template is a legacy view template. It was just doing the same thing. But as we can not simply remove and replace the template to keep backward compatibility ( if overridden from the theme ), I just re-used the new template in it.